apps/etterlatte-saksbehandling-ui/client/src/components/oppgavebenk/tildeling/VelgSaksbehandler.tsx

```diff
@@ -0,0 +1,164 @@
+  erRedigerbar: boolean
+  versjon: number | null
+  type: Oppgavetype
+}
+
+export const VelgSaksbehandler = ({
+  saksbehandler,
+  saksbehandlereIEnhet,
+  erRedigerbar,
+  oppgaveId,
+  sakId,
+  type,
+  versjon,
+  oppdaterTildeling,
+}: Props): ReactNode => {
+  const innloggetSaksbehandler = useAppSelector((state) => state.saksbehandlerReducer.innloggetSaksbehandler)
+
+  const [openDropdown, setOpenDropdown] = useState<boolean>(false)
+
+  const [valgtSaksbehandler, setValgtSaksbehandler] = useState<Saksbehandler | undefined>(saksbehandler)
+
+  const [, tildelSaksbehandler] = useApiCall(tildelSaksbehandlerApi)
+  const [, fjernSaksbehandler] = useApiCall(fjernSaksbehandlerApi)
+  const [, byttSaksbehandler] = useApiCall(byttSaksbehandlerApi)
+
+  const onSaksbehandlerSelect = (saksbehandlerNavn: string, erValgt: boolean) => {
+    if (erValgt) {
+      const selectedSaksbehandler: Saksbehandler | undefined = saksbehandlereIEnhet.find(
+        (behandler) => behandler.navn === saksbehandlerNavn
+      )
+
+      if (selectedSaksbehandler) {
+        byttSaksbehandler(
+          { oppgaveId, type, nysaksbehandler: { saksbehandler: selectedSaksbehandler.ident!, versjon } },
+          (result) => {
+            oppdaterTildeling(oppgaveId, selectedSaksbehandler.ident, result.versjon)
+            setValgtSaksbehandler(saksbehandler)
+            setOpenDropdown(false)
+          },
+          (error) => console.log(error)
+        )
+      }
+    }
+  }
+
+  const onTildelTilMeg = () => {
+    tildelSaksbehandler(
+      { oppgaveId, type, nysaksbehandler: { saksbehandler: innloggetSaksbehandler.ident, versjon } },
+      (result) => {
+        oppdaterTildeling(oppgaveId, innloggetSaksbehandler.ident, result.versjon)
+        setValgtSaksbehandler({ navn: innloggetSaksbehandler.navn, ident: innloggetSaksbehandler.ident })
+        setOpenDropdown(false)
+      },
+      (error) => {
+        console.log(error)
+      }
+    )
+  }
+
+  const onFjernTildeling = () => {
+    fjernSaksbehandler(
+      { oppgaveId, sakId, type, versjon },
+      (result) => {
+        oppdaterTildeling(oppgaveId, null, result.versjon)
+        setValgtSaksbehandler(undefined)
+        setOpenDropdown(false)
+      },
+      (error) => console.log(error)
+    )
+  }
+
+  return (
+    <div>
+      {erRedigerbar ? (
+        <Dropdown open={openDropdown}>
+          <Button
+            as={Dropdown.Toggle}
+            icon={valgtSaksbehandler?.ident ? <PersonPencilIcon /> : <PersonPlusIcon />}
+            iconPosition="left"
+            size="small"
+            variant="tertiary"
+            onClick={() => setOpenDropdown(true)}
+          >
+            {valgtSaksbehandler?.navn
+              ? valgtSaksbehandler.navn === innloggetSaksbehandler.navn
+                ? `${valgtSaksbehandler.navn} (meg)`
+                : valgtSaksbehandler.navn
+              : 'Ikke tildelt'}
```

Comment:
Forslag for å lettere se hva som skjer istedenfor å nøste de. 

```suggestion
            {valgtSaksbehandler?.navn
                ? `${valgtSaksbehandler.navn}${valgtSaksbehandler.navn === innloggetSaksbehandler.navn ? '(meg)':''}`
              : 'Ikke tildelt'}
```